src/hooks/useTransactionData/useTransactionData.tsx

```diff
@@ -0,0 +1,23 @@
+import { useSearchParams } from 'react-router-dom';
+
+import { transactionDataKeys } from '../../types';
+
+function useTransactionData() {
+  const [searchParams] = useSearchParams();
+
+  const URLTransactionData: { [x: string]: any } = {};
+  transactionDataKeys.forEach((key: string) => {
+    const res = searchParams.get(key);
+    if (res) {
+      URLTransactionData[key] = res;
+    }
+  });
+  const URLContractType = searchParams.get('contractType');
+  const URLInteractionType = searchParams.get('interactionType');
+
+  // TODO: implement autofixes to type compatibility descrepencies ( if transaction data includes valid create ant keys for example, set the contract and interaction types appropriately)
```

Comment:
Let me see if I get this: you want to set a type based on a value. Can you explain where/how exactly would this type be used?

I'm leaving the syntax for custom type wards which may be useful here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
contract have individual functions - we have the REGISTRATION_INTERACTION enum and likewise for the ant interactions. We then have the RegistrationInteraction and AntInteraction types `(typeof REGISTRY_INTERACTION_TYPES)[keyof typeof REGISTRY_INTERACTION_TYPES];`. 

So - if the hook pulls a contract type of 'registry' but an interaction type of AntInteraction then that would be an incompatible contract-interaction type - even if functionName is named the same (because the props on that function name are probably different). 

So the validation would compare those values. Ideally the `functionName` from transaction data as well.

Reply:
I think this is what we talked about in the huddle 👍